src/internal.rs

```diff
@@ -928,89 +931,158 @@ impl BitDepthDependentType for AlPal {
     type T<BD: BitDepth> = [[[[BD::Pixel; 8]; 3]; 32]; 2]; /* [2 a/l][32 bx/y4][3 plane][8 palette_idx] */
 }
 
-#[derive(Clone, Copy)]
-#[repr(C)]
-pub struct Rav1dTaskContext_scratch_compinter_seg_mask {
+#[derive(Clone, Copy, FromZeroes, FromBytes, AsBytes)]
+#[repr(C, align(64))]
+pub struct ScratchCompinter {
     pub compinter: [[i16; 16384]; 2],
     pub seg_mask: [u8; 16384],
 }
 
-pub struct Lap;
+// Larger of the two between `ScratchCompinter` and `[BD::Pixel; 128 * 32]`.
+const SCRATCH_COMPINTER_SIZE: usize = mem::size_of::<ScratchCompinter>();
```

Comment:
Actually, I think it's fine.  Ideally we'd check this stuff all at compile time (I'm imagining  a `#[derive(SafeUnion)]` on a `union` that turns it into this kind of `struct` with `zerocopy` methods), but I think this is fine for and fully safe, the checks are just done at runtime.

Reply:
Yeah I opted not to do the max comparison since I could figure out the larger variants manually, and the `FromBytes` conversion would fail if we were accidentally using the smaller of the two. We can always clean this up later if we find that maintaining the safe unions is difficult.